app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageViewModel.kt

```diff
@@ -812,9 +817,31 @@ class NewMessageViewModel @Inject constructor(
     }
 
     private fun Draft.updateDraftAttachmentsFromLiveData() {
-        val updatedAttachments = attachmentsLiveData.valueOrEmpty().map { uiAttachment ->
-            // If a localAttachment has the same `uploadLocalUri` than a UI one, it means it represents the same Attachment.
-            val localAttachment = attachments.filter { it.uploadLocalUri == uiAttachment.uploadLocalUri }
+
+        val uiAttachments = attachmentsLiveData.valueOrEmpty()
+
+        /**
+         * If :
+         * - we are in FORWARD mode,
+         * - all Attachments have no `uploadLocalUri` (meaning they are all from the original forwarded Message),
+         * - there quantity is the same in UI and in Realm,
+         * Then it means the Attachments list hasn't be edited by the user, so we have nothing to do here.
+         */
+        if (draftMode == DraftMode.FORWARD &&
+            uiAttachments.all { it.uploadLocalUri == null } &&
```

Comment:
Having to constantly check throughout the app if we an `uploadLocalUri` or not to actually get another derived information is pretty hard to understand when reading the code with no context. Can we create some kind of property with a meaningful name that does this very check behind the scene? 

On a side note, this could go in a second PR if it impacts multiple places in the code I think

Reply:
Done in #1930.